public class org.apache.batik.dom.util.SAXIOException extends java.io.IOException
{
    protected org.xml.sax.SAXException saxe;

    public void <init>(org.xml.sax.SAXException)
    {
        org.apache.batik.dom.util.SAXIOException r0;
        java.lang.String $r2;
        org.xml.sax.SAXException r1;

        r0 := @this: org.apache.batik.dom.util.SAXIOException;

        r1 := @parameter0: org.xml.sax.SAXException;

        $r2 = virtualinvoke r1.<org.xml.sax.SAXException: java.lang.String getMessage()>();

        specialinvoke r0.<java.io.IOException: void <init>(java.lang.String)>($r2);

        r0.<org.apache.batik.dom.util.SAXIOException: org.xml.sax.SAXException saxe> = r1;

        return;
    }

    public org.xml.sax.SAXException getSAXException()
    {
        org.apache.batik.dom.util.SAXIOException r0;
        org.xml.sax.SAXException $r1;

        r0 := @this: org.apache.batik.dom.util.SAXIOException;

        $r1 = r0.<org.apache.batik.dom.util.SAXIOException: org.xml.sax.SAXException saxe>;

        return $r1;
    }

    public java.lang.Throwable getCause()
    {
        org.apache.batik.dom.util.SAXIOException r0;
        org.xml.sax.SAXException $r1;

        r0 := @this: org.apache.batik.dom.util.SAXIOException;

        $r1 = r0.<org.apache.batik.dom.util.SAXIOException: org.xml.sax.SAXException saxe>;

        return $r1;
    }
}
